ads/imagead.md

```diff
@@ -0,0 +1,137 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Imagead
+
+Imagead does not represent a specific network. Rather, it provides a way for 
+a site to display simple image ads on a self-service basis. You must provide
+your own ad server to deliver the ads in json format as shown below.
+
+## Examples
+
+### Single ad
+
+```html
+<amp-ad width=300 height=250
+    type="adimage"
+    data-slot="1"
+    data-url="https://mysite/my-ad-server"
+>
+</amp-ad>
+<!-- The ad server will be called with the URL https://mysite/my-ad-server?s=1 -->
+```
+
+### Two ads with different slots
+
+```html
+<amp-ad width=300 height=250
+    type="adimage"
+    data-slot="1"
+    data-url="https://mysite/my-ad-server"
+>
+</amp-ad>
+<amp-ad width=400 height=300
+    type="adimage"
+    data-slot="2"
+    data-url="https://mysite/my-ad-server"
+>
+</amp-ad>
+<!-- The ad server will be called with the URL https://mysite/my-ad-server?s=1,2 -->
+```
+
+### Ads from different ad servers
+```html
+<amp-ad width=300 height=250
+    type="adimage"
+    data-slot="slot-name-a"
+    data-url="https://mysite/my-ad-server"
+>
+</amp-ad>
+<amp-ad width=400 height=300
+    type="adimage"
+    data-slot="slot-name-b"
+    data-url="https://mysite/my-ad-server"
+>
+</amp-ad>
+<amp-ad width=300 height=250
+    type="adimage"
+    data-slot="123"
+    data-url="https://my-other-site/my-other-ad-server"
+>
+</amp-ad>
+<!-- Two ad server calls will be made: -->
+<!-- The first:  https://mysite/my-ad-server?s=slot-name-a,slot-name-b -->
+<!-- The second: https://my-other-site/my-other-ad-server?s=123 -->
+```
+
+
+
+## Supported parameters
+
+### data-url (mandatory)
+
+This must be starting with `https://`, and it must be the address of an ad
+server returning json in the format defined below.
+
+### data-slot (mandatory)
```

Comment:
The slot isn't just a question of batching the request: it actually identifies whole different categories of ads for different places on the page - for example, it distinguishes a leaderboard ad from a middle-of-content ad from a sidebar ad (and thus defines the size of the ad). It can also be guaranteed to be different for different sites if you have more than one site using the same ad server.

It doesn't hurt to make it optional, since the user *might* have a very simple system with only one slot. So I'll do that.